### PR TITLE
Fix for subemotes problem 

### PR DESCRIPTION
**Pull request by TroyDota**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

### CODE REVIEW

File: pajbot/apiwrappers/twitchemotesapi.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            resp = self.get(["channels", channel_id])
+            plans = resp["plans"]
+            if len(plans) <= 0:
+                log.warning(f"No subscription plans found for channel {channel_name}")
+                return [], [], []
+
+            # plans["$4.99"] is tier 1
+            # plans["$9.99"] is tier 2
+            # plans["$24.99"] is tier 1
+            ret_data = [[], [], []]
```
Comment (RAnders00) [inline pajbot/apiwrappers/twitchemotesapi.py]: Just nitpicking but can‘t we declare this as `([], [], [])` to save the call to `tuple()` later?

File: pajbot/apiwrappers/twitchemotesapi.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        except HTTPError as e:
+            if e.response.status_code == 404:
+                log.warning(f"No sub emotes found for channel {channel_name}")
+                return [], [], []
+            else:
+                raise e
+
+    def get_channel_emotes(self, channel_id, channel_name, force_fetch=False):
+        return self.cache.cache_fetch_fn(
+            redis_key=f"api:twitch:legacy:channel-emotes:{channel_name}",
```
Comment (RAnders00) [inline pajbot/apiwrappers/twitchemotesapi.py]: This redis key should change

File: pajbot/managers/emote.py [hunk 3]
```diff
@@ -82,7 +83,7 @@ class TwitchEmoteManager(GenericChannelEmoteManager):
     def __init__(self, twitch_v5_api, twitch_legacy_api):
         self.api = twitch_v5_api
         self.legacy_api = twitch_legacy_api
-
+        self.twitchemotesapi = TwitchEmotesAPI(RedisManager.get())
```
Comment (RAnders00) [inline pajbot/managers/emote.py]: I would call it `twitchemotes_api` or `twitch_emotes_api`

File: pajbot/managers/emote.py [hunk 4]
```diff
@@ -95,14 +96,14 @@ def channel_emotes(self):
 
     def load_channel_emotes(self):
         streamer = StreamHelper.get_streamer()
-        self.tier_one_emotes, self.tier_two_emotes, self.tier_three_emotes = self.legacy_api.get_channel_emotes(
-            streamer
+        self.tier_one_emotes, self.tier_two_emotes, self.tier_three_emotes = self.twitchemotesapi.get_channel_emotes(
+            StreamHelper.get_streamer_id(), streamer
```
Comment (RAnders00) [inline pajbot/managers/emote.py]: Getting streamer and streamer ID should be in the same fashion so either inline the `streamer` variable or extract the function call to a variable

**Comment by RAnders00** [comment]
@TroyDota I would definitely like to merge these changes. Why did you close this pull request? Once the small issues I commented on are fixed we can merge this.

The two commits that you added (I presume accidentally) can be reverted without any problems, if needed.

**Comment by TroyDota** [comment]
Sorry i was really busy with another bot, discord this time. Tier 3 and tier 2 sub roles for discord users.

**Comment by TroyDota** [comment]
Any more changes? I dont think the redis address is being used at all to store the emote data? not fully committed to this pr rn but ill get better sorry

**Comment by TroyDota** [comment]
I removed the ones that are not relevant before i commented.

**Comment by alazymeme** [comment]
^ Don't mind this, was fucking around with something.

Pls make sure you test before merging/making a final pr. This commit broke my bot and I had to revert commits.

**Comment by RAnders00** [comment]
Well it was not released to stable yet
From my understanding master is not meant to always work, it can be a development snapshot too

**Comment by TroyDota** [comment]
Did this not work? I am currently running this code on my bot and it's working fine?

**Pull request merged**